server/gameconnection.py

```diff
@@ -210,41 +219,32 @@ async def handle_desync(self, *_args):  # pragma: no cover
         self.game.desyncs += 1
 
     async def handle_game_option(self, key, value):
-        if key == 'Victory':
-            self.game.gameOptions['Victory'] = Victory.from_gpgnet_string(value)
-        elif key in self.game.gameOptions:
-
-            """
-            This block about AIReplacement is added because of a mistake in the FAF game patch code
-            that makes "On" and "Off" be "AIReplacementOn" and "AIReplacementOff". The code
-            below removes that extra statement to make it a simple "On" "Off".
-            This block can be removed as soon as the game sends "On" and "Off" instead of
-            "AIReplacementOn" and "AIReplacementOff" to the server as game options.
-            https://github.com/FAForever/fa/issues/2610
-            """
-            if key == "AIReplacement":
-                value = value.replace("AIReplacement", "")
+        if not self.is_host():
+            return
```

Comment:
All of these `if not self.is_host()` checks might work well in  a `@host_only` decorator.
(Although it's slightly cumbersome to give them access to `self`, see [here](https://medium.com/@vadimpushtaev/decorator-inside-python-class-1e74d23107f6).)

Reply:
I decided not to do a decorator because the method signatures of the handlers are actually used for command validation. Adding a decorator would change the function signature of every handler to the same thing `func(self, *args)` which I don't want.